rskj-core/src/main/java/org/ethereum/core/ImmutableTransaction.java

```diff
@@ -10,12 +12,12 @@ public ImmutableTransaction(byte[] rawData) {
 
     @Override
     public void sign(byte[] privKeyBytes) {
-        throw new ImmutableTransactionException("trying to sign");
+        throw new ImmutableTransactionException(String.format("trying to sign tx=%s", Hex.toHexString(this.getHash())));
     }
 
     @Override
     public void setGasLimit(byte[] gasLimit) {
-        throw new ImmutableTransactionException("trying to set gas limit");
+        throw new ImmutableTransactionException(String.format("trying to set gas limit tx=%s", Hex.toHexString(this.getHash())));

```

Comment:
The String.format FORCES the execution and concatenation of the references expressions. Log message with {} only invokes .toString when the message level is on.

Reply:
Here I'm adding some info on the Exception, I was tryng to avoid simple concatenation